praw/models/reddit/submission.py

```diff
@@ -78,6 +78,89 @@ def select(self, flair_template_id: str, text: Optional[str] = None):
         self.submission._reddit.post(url, data=data)
 
 
+class SubmissionModerationFlair(SubmissionFlair):
+    """Provides a set of functions allowing moderators to set post flairs."""
+
+    def __call__(self, text: str = "", css_class: str = ""):
+        """Set flair for the submission.
+
+        :param text: The flair text to associate with the Submission (default:
+            '').
+        :param css_class: The css class to associate with the flair html
+            (default: '').
+
+        This method can only be used by an authenticated user who is a
+        moderator of the Submission's Subreddit.
+
+        Example usage:
+
+        .. code-block:: python
+
+           submission = reddit.submission(id='5or86n')
+           submission.mod.flair(text='PRAW', css_class='bot')
+
+        """
+        data = {
+            "css_class": css_class,
+            "link": self.submission.fullname,
+            "text": text,
+        }
+        url = API_PATH["flair"].format(subreddit=self.submission.subreddit)
+        self.submission._reddit.post(url, data=data)
+
+    def choices(
```

Comment:
Do we need this method? It appears that one could simply use `subreddit.flair.link_templates`.

Reply:
They could, but initially I subclassed submission.flair.